### PR TITLE
Strict record handling

### PR DESCRIPTION
**Pull request by derjust**
Hi!
I have the requirement to fail if the records are incomplete. Up to now, fast-csv is quite polite and just consideres an incomplete line as the first field/property and continues.

A workaround wouuld be to check in the validate function if the row is incomplete. but based on the amount of fields, it is very complex to identify a record which may have some fields left empty (which might be valid) and some literally missing.

Therefore introduced a new property (on default, the existing behavior remains unchanged) to emit data-invalid events on incomplete rows.

### CODE REVIEW

**Comment by derjust** [comment]
Added testcase to describe the intended behavior and updated to the latest HEAD

File: test/parsing.test.js [hunk 1]
```diff
@@ -985,6 +992,27 @@ it.describe("fast-csv parsing", function (it) {
...
+        csv
+            .fromPath(path.resolve(__dirname, "./assets/test25.csv"), {headers: true, strictColumnHandling: true})
+            .on("data", function (data) {
+                actual.push(data);
+            })
+            .on("data-invalid", function(actual) {
+                assert.deepEqual(actual, expected25_invalid);
+                reachedInvalid = true;
+            })
+            .on("error", function(e){console.log(e)})
```
Comment (doug-martin) [inline test/parsing.test.js]: We should error on here you can just do `on("error", next)`

**Comment by doug-martin** [comment]
Other than my one comment this looks good Ill merge once fixed.

Thanks again
-Doug

File: test/parsing.test.js (hunk 1 above)
Comment (derjust) [inline test/parsing.test.js]: Fixed with e859b69
Thanks

**Pull request merged**